File: tests/output/jpa/src/main/javagen/restaurant/jpa_feign/dtos/restaurant/FactureItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_feign.dtos.restaurant;

import jakarta.annotation.Generated;
import jakarta.validation.constraints.NotNull;

/**
 * Détail d'une facture.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public record FactureItem(
	@NotNull Integer id,
	@NotNull Integer commandeId
) {

	/**
	 * Enumération des champs de la classe {@link restaurant.jpa_feign.dtos.restaurant.FactureItem FactureItem}.
	 */
	public enum Fields {
		ID(Integer.class),
		COMMANDE_ID(Integer.class);

		private final Class<?> type;

		Fields(Class<?> type) {
			this.type = type;
		}

		/**
		 * Getter for type.
		 *
		 * @return value of {@link #type type}.
		 */
		public Class<?> getType() {
			return this.type;
		}
	}
}

File: tests/output/focus/src/model/restaurant/facture-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {e, entity, EntityToType} from "@focus4/entities";
import {DO_ID, DO_ID_2} from "../../domains";

export type FactureItem = EntityToType<FactureItemEntityType>;
export type FactureItemEntityType = typeof FactureItemEntity;

export const FactureItemEntity = entity({
    id: e.field(DO_ID, f => f
        .label("restaurant.facture.id")
    ),
    commandeId: e.field(DO_ID_2, f => f
        .label("restaurant.facture.commandeId")
    )
});
